libs/contract-core/src/index_mocked.ts

```diff
@@ -7,44 +7,83 @@ const genRanHex = (size: number) => [...Array(size)].map(() => Math.floor(Math.r
 function delay(ms: number) {
   return new Promise(resolve => setTimeout(resolve, ms));
 }
-export class TheCoin implements Pick<Src.TheCoin, 'exactTransfer'|'balanceOf'|'certifiedTransfer'>{
+let nonce = 12;
+let confirmations = 1;
+export class TheCoin implements Pick<Src.TheCoin, 'exactTransfer' | 'balanceOf' | 'certifiedTransfer'>{
 
-  genReceipt = (opts?: any) => Promise.resolve({
-    wait: () => { },
-    hash: `0x${genRanHex(64)}`,
-    ...opts,
-  } as any as ContractTransaction)
+  // prefix is (e) for exactTransfer and (c) for certifiedTransfer, and (0) for other
```

Comment:
This is becoming very convoluted, and is mostly unnecessary.

We can instead standardize on a mocked provider and then just use the regular contract

https://github.com/thecointech/thecoin/issues/376